subtests/docker_cli/attach/attach.py

```diff
@@ -36,10 +36,34 @@ def wait_interactive_cmd(self):
                      % wait)
         time.sleep(wait)
 
+    def image_exist(self, image_name):
+        di = DockerImages(self)
+        image_list = DockerImages.get_dockerimages_list(di)
+        exist_status = DockerImages.filter_list_full_name(image_list,
+                                                          image_name)
+        return exist_status
+
+    def pull_image(self, image_name):
+        print "pull_image"
+        dkrcmd = AsyncDockerCmd(self, 'pull', [image_name],
+                                self.config['docker_timeout'],
+                                verbose=True)
+        self.loginfo("Executing background command: %s" % dkrcmd)
+        dkrcmd.execute()
+        while not dkrcmd.done:
+            self.loginfo("Pulling...")
+            time.sleep(3)
+        result = dkrcmd.wait()
+        self.failif(result.exit_status != 0,
+                    "Fail to download image %s"
+                    % image_name)
+
     def initialize(self):
         super(attach_base, self).initialize()
         self.sub_stuff['subargs'] = self.config['run_options_csv'].split(',')
         fin = DockerImage.full_name_from_defaults(self.config)
+        if self.image_exist(fin) == []:
+            self.pull_image(fin)
```

Comment:
Hmmm, in fact, maybe this should only be a *pretest* that only runs once at beginning.  Docker-autotest already has common ``clean_all()`` code to protect this default FQIN image.  Then if docker breaks and the image is deleted, the next test will correctly fail.  What do you think?  Ya, I don't think [including intra-into-pre](https://github.com/autotest/autotest-docker/compare/autotest:master...cevich:preintra) is good idea.